pycheribuild/projects/cross/cheribsd.py

```diff
@@ -1476,56 +1776,39 @@ def _build_and_install_kernel_binaries(self, build_cheribsd: BuildCHERIBSD, kern
-        return build_cheribsd.kernel_config
-
-    @classmethod
-    def get_installed_kernel_path(cls, caller: SimpleProject, config: CheriConfig = None,
-                                  cross_target: CrossCompileTarget = None) -> Path:
-        return cls.installed_kernel_for_config(caller, cls.get_kernel_config(caller, cross_target), config,
-                                               cross_target)
-
-    @classmethod
-    def get_installed_benchmark_kernel_path(cls, caller: SimpleProject, config: CheriConfig = None,
-                                            cross_target: CrossCompileTarget = None) -> Path:
-        return cls.installed_kernel_for_config(caller, cls.get_kernel_config(caller, cross_target), config,
-                                               cross_target, prefer_benchmark_kernel=True)
-
-    @staticmethod
-    def installed_kernel_for_config(caller: SimpleProject, kernconf: str, config: CheriConfig = None,
-                                    cross_target: CrossCompileTarget = None, prefer_benchmark_kernel=False) -> Path:
-        if config is None:
-            config = caller.config
-        if cross_target is None:
-            cross_target = caller.crosscompile_target
-        guess = config.cheribsd_image_root / (
-                "kernel" + cross_target.build_suffix(config, include_os=False) + "." + kernconf)
-        if prefer_benchmark_kernel:
-            for benchmark_suffix in ("-BENCHMARK", "-NODEBUG", "_BENCHMARK", "_NODEBUG"):
-                benchmark_guess = guess.with_name(guess.name + benchmark_suffix)
-                if benchmark_guess.exists():
-                    return benchmark_guess
-        return guess
+    def default_kernel_config(self, platform: ConfigPlatform = None, **filter_kwargs) -> str:
+        if platform is None:
+            platform = self.get_default_kernel_platform()
+        kABI = filter_kwargs.pop("kABI", self.get_default_kernel_abi())
+        filter_kwargs["mfsroot"] = True
+        if self.caprevoke_kernel:
+            filter_kwargs["caprevoke"] = True
+        config = CheriBSDConfigTable.get_default(self.crosscompile_target, platform, kABI, **filter_kwargs)
+        return config.kernconf
+
+    def get_kernel_configs(self, **filter_kwargs) -> "typing.Sequence[str]":
+        xtarget = self.crosscompile_target
+        default_kABI = self.get_default_kernel_abi()
+        other_kABI = KernelABI.PURECAP if default_kABI != KernelABI.PURECAP else KernelABI.HYBRID
+        platforms = self.get_default_kernel_platform()
+        if self.build_fpga_kernels:
+            platforms |= ConfigPlatform.fpga_platforms()
+        configs = CheriBSDConfigTable.get_configs(xtarget, platforms, default_kABI,
+                                                  mfsroot=True)
+        configs += CheriBSDConfigTable.get_configs(xtarget, platforms, default_kABI,
+                                                   mfsroot=True, benchmark=True)
+        if self.build_purecap_kernels:
```

Comment:
This is because when the default kernel is already purecap, the build-purecap-kernels flag adds the hybrid kernels to the build instead. This was changed originally after commenting with @jrtc27

Reply:
Ah I missed that discussion. Maybe we should rename the field to something like "build_alternative_abi_kernels"?